### ISSUE TITLE
Issues retiring worker

### ISSUE THREAD

**Issue by akrajewski** [ahelal/ansible-concourse#78]
Hello,

I have upgraded Concourse from v3.14.1 to v4.2.2 which generally went smooth. However, it seems that the logic for retiring the worker has been broken.

When I execute `systemctl stop concourse-worker`, the command freezes and reports failure to retire although it seems that in the background everything executes fine (i.e. worker is deleted from Concourse).

I have done some digging and discovered that the retire_worker routine in the init script is checking for a certain log message:

```
 if grep "retire-worker.worker-not-found" "${temp_file}" > /dev/null 2>&1; then
                log "Officially retired ${CONCOURSE_NAME} worker"
                rm "${temp_file}"
                return 0
```

This log message seems to not be present in 4.2.2:

```
root@machine:/opt/concourseci/bin# ./concourse retire-worker
{"timestamp":"1549980736.191713333","source":"retire-worker","message":"retire-worker.beacon.setting-up-beacon-runner","log_level":0,"data":{"session":"1"}}
{"timestamp":"1549980736.191801548","source":"retire-worker","message":"retire-worker.beacon.retire-worker","log_level":0,"data":{"session":"1"}}
{"timestamp":"1549980736.191845655","source":"retire-worker","message":"retire-worker.beacon.command-to-run","log_level":0,"data":{"cmd":"retire-worker","session":"1"}}
{"timestamp":"1549980736.223448753","source":"retire-worker","message":"retire-worker.beacon.beacon-client.keepalive.ok","log_level":0,"data":{"session":"1.1.1"}}
{"timestamp":"1549980736.223629475","source":"retire-worker","message":"retire-worker.beacon.ssh-forward-config","log_level":0,"data":{"bcForwardAddrRemote":"","gardenForwardAddrRemote":"","session":"1"}}
```

This is probably due to https://github.com/concourse/bin/issues/34 being fixed in Concourse. TSA previously had hardcoded DEBUG log level which caused the logs to appear in retire-worker output.(and indeed - according to my quick check, changing the CONCOURSE_TSA_LOG_LEVEL to 'debug' on the web node causes the log message to re-appear).

I could submit a PR to resolve this. Thinking about checking the return code of `concourse retire-worker` instead of looking for a certain debug log pattern (which can easily change from version to version). Not sure if there is any specific reason why the log-based approach has been chosen?

**Comment by akrajewski** [comment]
After further analysis I have noticed

**Comment by akrajewski** [comment]
After further analysis it seems that the return code is already checked. Therefore I guess we could maybe just wait for graceful exit of the concourse worker process? Just submitted a PR to address this: https://github.com/ahelal/ansible-concourse/pull/79

**Comment by akrajewski** [comment]
Closing, should be resolved by PR #79.
